### DISCUSSION TITLE
[Question]: Event listener to visibility(eye) icon in the Layer Manager

### DISCUSSION THREAD

**Discussion by sathyanarayananaCES** [GrapesJS/grapesjs#3635] [Q&A]
I would like to listen to the eye icon in the layer manager from the editor, based on which, I need to write the custom styles. Is there any way we can do that?

I got the code, from [another issue](https://github.com/artf/grapesjs/issues/3154#issuecomment-733693886), I would like to extend the toggleVisibility method from here. Is it possible? if so, how can we do that?

Secondly, I would like to write custom media queries to the selected element on change of the visibility icon, how can it be done in the best way?

Thank you

**Comment by ronaldohoch** [comment]
Hello!
For your first question, i'm using this code:
```js
    editor.on("run:preview", () => {
      console.log("run:preview");
    });
    editor.on("stop:preview", () => {
      console.log("stop:preview");
    });
```

For the second question, the preview code removes the `gjs-selected` from the element, so i think you can't do that.
